arangod/Agency/AgentConfiguration.h

```diff
@@ -252,6 +257,11 @@ struct config_t {
     _supervisionDelayFailedFollower = d;
   }
 
+  /// @brief set Supervision FailedLeader adds follower flag
+  void setSupervisionFailedLeaderAddsFollower(bool f) {
+    _supervisionFailedLeaderAddsFollower = f;
```

Comment:
It is a bit shady that reading attributes from the configuration object is done under a read-lock and writes to the configuration object are done under to write-lock, except for these setters...
I think to be consistent, all these setters should acquire the write-lock as well.

Reply:
Agreed.